### ISSUE TITLE
Intance of OrdersRepositoryInterface gets not called on "SetOrderState" Actions

### ISSUE THREAD

**Issue by brainspin-max** [billbeeio/custom-shop-php-sdk#8]
Instances of OrdersRepositoryInterface do not get called on "SetOrderState" Actions. The Problem seems to be that the deserialization of the billbee request fails and the orderId cannot be found.

I logged the following billbee request that the sandbox made to out webapp:

```json
{"OrderId":"4882","Date":"2022-02-16 11:09:25","TrackingCode":"","NewStateId":"Versendet","Action":"SetOrderState","Key":"<our-shop-key>"}
```

The request fails inside the `Billbee\CustomShopApi\RequestHandler\OrderRequestHandler` at line 114 because no OrderId could be found. Non of the OrderRepositories get called.

When i debug the request before its been send to the billbee api, i can see that the params are hand to the billbee sdk inside the request GET Parameters inside the uri. The OrderRequestHandler seems to tryin to get the params from the request body which seems to be empty.

Propably the error would get fixed by changing the `OrderRequestHandler.php` `handle` function to pass the queryArgs along with the request object and inside the `setOrderState` check inside the queryArgs and not inside the request. 

As i see some other funktions do the same, so maybe this problem effects some other actions as well.

**Comment by brainspin-max** [comment]
@devtronic Ive done some additional debugging inside the OrderRequestHandler. As seen above and in the following request, billbee sends a string with the name of the new state inside the NewStateId parameter. The `setOrderState()` function on the other hand expects an integer value as the new state id at line 122.

@devtronic  Could it be, that billbee just sends the **name** instead of the **id** parameter inside the `NewStateId` and so causes the setOrderState function to fail?

**Comment by marconett** [comment]
The issue is basically the same as #7 and was fixed in part in pull request #2 (which is still not merged).

This SDK expects a JSON body, for the `SetOrderState` request, but the API sends `application/x-www-form-urlencoded`. #2 fixes this, but there's another issue:

They altered the postman file to mirror the actual POST that Billbee sends:
https://github.com/billbeeio/custom-shop-api-postman/commit/master#diff-15fb4c5391937118691038193caabce6d98c63c8d66ad196362aec97ba413d52L138

But they did NOT alter this SDK to reflect that change: https://github.com/billbeeio/custom-shop-php-sdk/blob/master/src/RequestHandler/OrderRequestHandler.php#L117

`NewStateId` is still expected to be an integer, when in fact, Billbee sends a string. Changing `NewStateId` to `NewStateTypeId` in that line should fix the problem.

I will be fixing this in my branch https://github.com/marconett/custom-shop-php-sdk.

**Comment by brainspin-max** [comment]
@marconett thanks for the insights and the hint with the form request, which explains allot. At the end i ended up not using the sdk at all and just call my OrderRepository directly with the provided Data. In my case in Laravel this looks like the following:

```php
// check the key if its valid by copying the content of the KeyAuthenticator and modify if for your framework

if(
  $request->input('Action') == 'SetOrderState' && 
  $request->input('OrderId') && 
  $request->input('NewStateTypeId')
){
  $orderRepository = new OrderRepository;
  $orderRepository->setOrderState($request->input('OrderId'), $request->input('NewStateTypeId'), null);
  return response()->json([], 201);
}
```

The benifit you get is that this way its more predictable and it doesnt matter if the request is a form or json request. I close the issue for now, because its solved for me and there allready issues that relate to this.

**Comment by brainspin-max** [comment]
I close the issue for now, because the problems are addresses in other issues and there allready pull requests.

**Comment by marconett** [comment]
this specific problem is not addressed in the pull request or the other issue.

i'd suggest keeping it open.

**Comment by devtronic** [comment]
Fixed in 2.1.1 (https://packagist.org/packages/billbee/custom-shop-api#v2.1.1)
